compiler/codegen/OMRRegisterIterator.hpp

```diff
@@ -22,36 +22,61 @@
 #ifndef OMR_REGISTER_ITERATOR_INCL
 #define OMR_REGISTER_ITERATOR_INCL
 
-/*
- * The following #define and typedef must appear before any #includes in this file
- */
-#ifndef OMR_REGISTER_ITERATOR_CONNECTOR
-#define OMR_REGISTER_ITERATOR_CONNECTOR
-namespace OMR { class RegisterIterator; }
-namespace OMR { typedef OMR::RegisterIterator RegisterIteratorConnector; }
-#endif
-
 #include "env/TRMemory.hpp"  // for TR_Memory, etc
+
+namespace TR { class Machine; }
 namespace TR { class Register; }
 
 namespace OMR
 {
 
-/*
- * This interface must be implemented by platform-specific descendents,
- * using either List and ListIterator, or some cheaper alternatives.
+/**
+ * \brief
+ * An implementation of the iterator pattern to allow users (for example RAS)
+ * traverse the platform-specific list of a certain kind of registers (GPR,
+ * FPR, VRF, etc.)
  */
-class OMR_EXTENSIBLE RegisterIterator
+class RegisterIterator
```

Comment:
One byproduct of this change is that the class is no longer extensible. I don't particularly see a problem with this and it can be extended at a later point if needed, but wanted to explicitly point this out to other committers.

Reply:
In issue #2645 @0xdaryl said:

> It's also not clear to me why this was made an extensible class in the first place. It's hardly a class I would think one would want to extend.

So, as I get, there is no reason to mark the class as `OMR_EXTENSIBLE`.